projects/packages/my-jetpack/_inc/components/info-tooltip/types.ts

```diff
@@ -0,0 +1,210 @@
+/* eslint-disable jsdoc/check-indentation */
+import type { ReactNode, MutableRefObject, SyntheticEvent } from 'react';
+
+type PositionYAxis = 'top' | 'middle' | 'bottom';
+type PositionXAxis = 'left' | 'center' | 'right';
+type PositionCorner = 'top' | 'right' | 'bottom' | 'left';
+
+type DomRectWithOwnerDocument = DOMRect & {
+	ownerDocument?: Document;
+};
+
+type PopoverPlacement =
+	| 'top'
+	| 'top-start'
+	| 'top-end'
+	| 'right'
+	| 'right-start'
+	| 'right-end'
+	| 'bottom'
+	| 'bottom-start'
+	| 'bottom-end'
+	| 'left'
+	| 'left-start'
+	| 'left-end'
+	| 'overlay';
+
+export type PopoverAnchorRefReference = MutableRefObject< Element | null | undefined >;
+export type PopoverAnchorRefTopBottom = { top: Element; bottom: Element };
+
+export type VirtualElement = Pick< Element, 'getBoundingClientRect' > & {
+	ownerDocument?: Document;
+};
+
+export type PopoverProps = {
```

Comment:
I'm guessing the Popover WordPress component doesn't have an exported types we could use instead? If not, how hard would it be to add them there? Only asking because the ideal scenario would be to have the component type itself, but I understand that may be more difficult than it's worth

Reply:
Yeah right, the Popover WordPress component doesn't have any exported types, unfortunately.  But yeah you're right, it would be nice if it did! 👍
I'll add a followup maintenance task to update the WordPress Popover component to export the Popover type, and then come back here to clean up the InfoTooltip types.